src/PDO/Peachpie.Library.PDO/PDOStatement.cs

```diff
@@ -443,6 +443,11 @@ public virtual PhpArray fetchAll(PDO_FETCH fetch_style = default, PhpValue fetch
                 }
             }
 
+            if (fetch_style.HasFlag(PDO_FETCH.FETCH_CLASS) && !fetch_argument.IsEmpty)
```

Comment:
we should not check for `fetch_argument.IsEmpty` here, `setFetchMode` should be called in any case once `fetch_style` is `FETCH_CLASS`